src/hypofuzz/interface.py

```diff
@@ -101,16 +104,32 @@ def pytest_collection_finish(self, session: pytest.Session) -> None:
                     flush=True,
                 )
                 continue
-
             # Wrap it up in a FuzzTarget and we're done!
             try:
-                # Skip state-machine classes, since they're not
-                if isinstance(item.obj, RuleBasedStateMachine.TestCase):
-                    target = partial(run_state_machine_as_test, item.obj)
+                if hasattr(item.obj, "_hypothesis_state_machine_class"):
+                    assert (
+                        extra_kw == {}
+                    ), "Not possible for RuleBasedStateMachine.TestCase to be parametrized"
+                    runTest = item.obj
+                    StateMachineClass = runTest._hypothesis_state_machine_class
+                    target = get_state_machine_test(  # type: ignore
+                        StateMachineClass,
+                        # runTest is a function, not a bound method, under pyest7.
+                        # I wonder if something about TestCase instantiation order
+                        # changed in pytest 8? Either way, we can't access
+                        # __self__.settings uder pytest 7.
+                        #
+                        # I am going to call this an acceptably rare bug for now,
+                        # because it should only manifest if the user sets a custom
+                        # database on a stateful test under pytest 7 (all non-db
+                        # settings are ignored by hypofuzz).
+                        settings=runTest.__self__.settings if pytest8 else None,
```

Comment:
Ugh, I really don't like this but I think it's probably the best option we practically have.

In the medium term, I'd like us to track all the tests-we-don't-collect, and have a dashboard page which lists them and explains why each was skipped (as well as optional terminal reporting?  hmm).

Reply:
yup I'd like that as well, should be doable nearish term. (our dashboard will probably turn into the standard "sidebar on the left which lists pages", and one page is a test overview or similar).

I will come back to this bug at some point, there are just many better things to do in the meantime!

Reply:
I think I found the relevant pytest change: https://github.com/pytest-dev/pytest/pull/12089. unsure if a fix will be easy

Reply:
let's make sure we pick frameworks etc which can be responsive / mobile-friendly; the dominant usecase will be on localhost but scaling beyond that is a goal.